addons/account/models/account_payment.py

```diff
@@ -56,8 +56,8 @@ class account_abstract_payment(models.AbstractModel):
     @api.one
     @api.constrains('amount')
     def _check_amount(self):
-        if not self.amount > 0.0:
-            raise ValidationError(_('The payment amount must be strictly positive.'))
+        if self.amount < 0:
+            raise ValidationError(_('The payment amount must be strictly non-negative.'))
```

Comment:
IMHO i think doing 'positive' would confuse user as we are now allowing zero digit and also 'non-negative' would be more precise then it